fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -559,6 +559,27 @@ class ShardManager {
 		dirtyShards->insert(it.value()->physicalShard);
 	}
 
+	void clear(KeyRef key) {
+		auto it = dataShardMap.rangeContaining(key);
+		if (!it.value()) {
+			return;
+		}
+		writeBatch->Delete(it->value()->physicalShard->cf, toSlice(key));
+		dirtyShards->insert(it.value()->physicalShard);
+	}
+
+	void clearRange(KeyRangeRef range) {
+		auto rangeIterator = dataShardMap.intersectingRanges(range);
+
+		for (auto it = rangeIterator.begin(); it != rangeIterator.end(); ++it) {
+			if (it.value() == nullptr) {
+				continue;
+			}
+			writeBatch->DeleteRange(it->value()->physicalShard->cf, toSlice(range.begin), toSlice(range.end));
+			dirtyShards->insert(it.value()->physicalShard);
+		}
+	}
+
```

Comment:
nit: Could all these logic be moved in to KeyValueStoreRocksDB::clear() and clearRange()? with the ShardManager providing the involving shards? Since these methods feel more belong to KeyValueStoreRocksDB.

Reply:
Yeah, this requires moving other data structures to KVS as well. I have some other PRs based on this one. I'll do a refactor to remove some shard manager APIs afterwards.